qcore.c

```diff
@@ -216,6 +216,10 @@ void QuerkSaveContext (void) {
 	asm("beq et, r0, SKIP_EA_DEC"); // interrupt is not external
 	asm("subi ea, ea, 4"); /* must decrement ea by one instruction for external
 							* interrupts, so that the instruction will be run */
+
+    register int generalRegister22Contents asm("r22");

```

Comment:
These two lines aren't needed. When you merge it, there will be a conflict. Just take the code in master, since calling trap (as you've done in qrequest.c) will call the_exception(), which will then call interrupt_handler.

Reply:
Removed those lines as requested.